src/models/HydroReservoir.jl

```diff
@@ -0,0 +1,128 @@
+#=
+This file exists here because HydroTurbine depends on it. Since autogenerated
+models sort imports alphabetically, HydroTurbine always appeared first, causing
+an import error.
+=#
+
+#! format: off
+
+"""
+    mutable struct HydroReservoir <: Component
+        name::String
+        available::Bool
+        storage_volume_limits::Union{Nothing, MinMax, TimeSeriesKey}
+        spillate_outflow_limits::Union{Nothing, MinMax, TimeSeriesKey}
+        inflow::Union{Float64, TimeSeriesKey}
+        outflow::Union{Float64, MinMax, TimeSeriesKey}
+        volume_targets::Union{Float64, MinMax, TimeSeriesKey}
+        travel_time::Union{Float64, MinMax, TimeSeriesKey}
+        services::Vector{Service}
+        ext::Dict{String, Any}
+        internal::InfrastructureSystemsInternal
```

Comment:
we need to generalize the data: storage_limits and targets, and add an enum if 1 = Head, and 2 = Volume for the data.

Reply:
I can explain tomorrow, but we are modeling head. That's the data that we have.